File: app/src/main/java/com/example/walkingdog_kotlin/Challenge/Model/Challenge.kt
package com.example.walkingdog_kotlin.Challenge.Model

import com.example.walkingdog_kotlin.ReformatDate

class Challenge () {
    var title: String = ""
    var comment:String = ""
    var period:String = ""
    var number:String = ""

    constructor(challenge: ChallengeModel): this() {
        this.title = challenge.title
        this.comment = challenge.content
        this.period = ReformatDate("MM월 dd일", challenge.start_date) + " ~ " +
                ReformatDate("MM월 dd일", challenge.end_date)
        this.number = challenge.population.toString()
    }
}

File: server/models/challenge.js
var mongoose = require('mongoose');
var Schema = mongoose.Schema;

const challengeSchema = new Schema({
    title: {
        type: String,
        required: true,
    },
    content: {
        type: String,  
    },
    start_date: {
        type: Date,
        required: true,
    },
    end_date: {
        type: Date,
        required: true,
    },
    date: {
        type: Number,
        default: 1,
    },
    population: {
        type: Number,
        required: true,
        default: 1,
    },
    populationLimit: {
        type: Number,
        required: true,
        default: 40,
    },
    producer: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true,
    },
    create_date: {
        type: Date,
        default: Date.now,
    },

})

module.exports = mongoose.model('Challenge', challengeSchema);